### PR TITLE
SyncBot: Add timestamp to message details

### PR DESCRIPTION
**Pull request by sqweelygig**
This is done as one step of preparation towards putting Jellyfish in as an endpoint

* Add time stamps to received messages

Change-Type: patch
Connects-To: #474

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [x] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #474

### CODE REVIEW

File: lib/services/messenger/translators/github.ts [hunk 1]
```diff
@@ -253,6 +253,7 @@ export class GithubTranslator extends TranslatorScaffold implements Translator {
 					hidden: _.isSet(metadata.hidden) ? metadata.hidden : false,
 					tags: thread.labels,
 					text: TranslatorScaffold.unixifyNewLines(text),
+					time: event.cookedEvent.data.created_at,
```
Comment (hedss) [inline lib/services/messenger/translators/github.ts]: Use `updated_at` rather than `created_at`. Any event that's updated will end up using the original creation date here, which may be far in the past.

File: tests/services/messenger/translators/discourse.spec.ts [hunk 2]
```diff
@@ -1,9 +1,8 @@
 /// <reference types="mocha" />
 import { expect } from 'chai';
 import * as _ from 'lodash';
-
+import * as crypto from 'crypto';
```
Comment (hedss) [inline tests/services/messenger/translators/discourse.spec.ts]: Why is this required? Can't see any changes in the code here that need it?

File: tests/services/messenger/translators/discourse.spec.ts [hunk 3]
```diff
@@ -16,6 +15,7 @@ describe('lib/services/messenger/translators/discourse.ts', () => {
 				hidden: false,
 				tags: [],
 				text: 'cde @test',
+				time: '2018-04-16T12:45:46+00:00',
```
Comment (hedss) [inline tests/services/messenger/translators/discourse.spec.ts]: Quick question on time here. Do timestamps get translated? How are these handled for timezone, or do they all get converted to Zulu?

Comment (sqweelygig) [reply tests/services/messenger/translators/discourse.spec.ts]: In Jellyfish these are stored as an ISO string
In the Jellyfish translator in intend on doing a `moment(time).toISOString()`
Therefore in the common message type these should be in any format that momentjs can parse.
I'm adding this to the interface declaration.

File: tests/services/messenger/translators/discourse.spec.ts (hunk 2 above)
Comment (hedss) [reply tests/services/messenger/translators/discourse.spec.ts]: My mistake, I hadn't seen it had moved from further down! :)
